modules/tinymce/src/core/test/ts/browser/keyboard/SpaceKeyTest.ts

```diff
@@ -16,12 +16,19 @@ describe('browser.tinymce.core.keyboard.SpaceKeyTest', () => {
 
   context('Space key around inline boundary elements', () => {
     it('Press space at beginning of inline boundary inserting nbsp', () => {
+      let inputEvents = 0;
       const editor = hook.editor();
       editor.setContent('<p>a <a href="#">b</a> c</p>');
+      editor.on('beforeinput input', (e) => {
+        // eslint-disable-next-line no-console
+        console.log(e.isComposing);
+        inputEvents++;
+      });
       TinySelections.setCursor(editor, [ 0, 1, 0 ], 0);
       editor.nodeChanged();
       TinyContentActions.keystroke(editor, Keys.space());
       TinyAssertions.assertSelection(editor, [ 0, 1, 0 ], 1, [ 0, 1, 0 ], 1);
+      assert.eq(2, inputEvents, 'Events not fired as expected');
```

Comment:
This doesn't test anything meaningful... as it doesn't test the order nor that you actually got a beforeinput and an  input. You could have ended up getting 2 beforeinput events instead, so please improve this to make sure events are firing in the correct order.